website/docs/guides/how-to/logging.md

```diff
@@ -1,89 +1,385 @@
+
+### 3. activate logging
+Finally, add the macro call `enable_frb_logging!();` in a **Rust** file that is part of your `rust_input` of your `flutter_rust_bridge.yaml` configuration, at any place outside of an item (e.g. function or struct). 
+You need to make it available via `use flutter_rust_bridge::enable_frb_logging;`.
+
+It needs to be there so the code generation is picking it up and generates the needed bridge code for connecting Rust and Dart for logging.
+
+### 4. customize the log function
+Per default any log output is written to `stdout`, via `println!` in Rust.
+You can provide your own log function, as described further down in `utilize a logging framework`.
+
+Note that this is needed, if you are developing a _Flutter_ application, as `stdout` is not shown.
+
+## How to use it
+
+All logs will be printed to `stdout`.
+
+You can issue log statements in Rust and Dart by following the usual way done in the respective 3rd party packages:
+
+In **Rust** simply call `log::info!()` (or any of the [log crate's log levels](https://docs.rs/log/latest/log/enum.Level.html)).
+
+In **Dart** you first need to initialise logging with a call to `final LOGGER = FRBLogger.initLogger()`.
+If you do this call more than once an Exception is thrown.
+This not only sets up logging for the Rust side, it returns a logger you can use to record log statements as well.
+
+Doing this in `main.dart`, preferably as a global variable `final LOGGER = FRBLogger.initLogger();`, is recommended, so no Rust log statement is executed before this setup.
+
+To get the logger handle after the initialization you can call `final LOGGER = FRBLogger.getLogger()`.
+This is in turn calling `FRBDartLogger.getLogger()`, which you can call directly instead (they are effectively the same).
+
+In both calls (`.initLogger()` and `.getLogger()`) you can provide a `LoggerName` (which defaults to "FRBLogger").
+
+To issue a log statement, you then call `LOGGER.info('Hello world');` (or other log levels) on that instance.
+Note that in **Rust** code you use the [log crate's log levels](https://docs.rs/log/latest/log/enum.Level.html), while in **Dart** code you use a mixture of those and [Dart's logging package equivalents](https://pub.dev/documentation/logging/latest/logging/Level-class.html).
+
+All of Dart's logging package log levels are available, but the more exotic ones have been replaced with the more common levels used in Rust:
+- FINE -> trace
+- CONFIG -> debug
+- SEVERE -> error
+- SHOUT -> fatal
+
+While Rust's log crate captures the module, file and line number of the log statement, Dart's logging package does not. 
+Here it is idiomatic to get a new handle of the logger in each file, and give it a name that tells you where your log statement is originating from.
+Usually one calls `final OTHER_LOGGER = Logger("other_logger");` from the library directly, but again, use `final OTHER_LOGGER = FRBLogger.getLogger("other_logger");` instead.
```

Comment:
nit: curious what does this do - it seems we only have one "logger that logs from rust"